lib/cgroup/cgroup.go

```diff
@@ -326,20 +327,36 @@ func (s *Service) unmount() error {
 	return nil
 }
 
+type fileHandle struct {
+	CgroupID uint64
+}
+
 // ID returns the cgroup ID for the given session.
 func (s *Service) ID(sessionID string) (uint64, error) {
+	var fh fileHandle
 	path := path.Join(s.teleportRoot, sessionID)
 
-	cpath := C.CString(path)
-	defer C.free(unsafe.Pointer(cpath))
+	// Call the "name_to_handle_at" syscall directly (unix.NameToHandleAt is a
+	// thin wrapper around the syscall) instead of calling the glibc wrapper.
+	// This has to be done to support older versions of glibc (like the one
+	// CentOS 6 ships with) which don't have the "name_to_handle_at" wrapper.
+	//
+	// Note that unix.NameToHandleAt is slightly more than a thin wrapper, it
+	// calls "name_to_handle_at" in a loop to get the correct size of the
+	// returned "f_handle" value. See the below link for more details.
+	//
+	// https://github.com/torvalds/linux/commit/f269099a7e7a0c6732c4a817d0e99e92216414d9
+	handle, _, err := unix.NameToHandleAt(unix.AT_FDCWD, path, 0)
+	if err != nil {
+		return 0, trace.Wrap(err)
+	}
 
-	// Returns the cgroup ID of a given path.
-	cgid := C.cgroup_id(cpath)
-	if cgid == 0 {
-		return 0, trace.BadParameter("cgroup resolution failed")
+	err = binary.Read(bytes.NewBuffer(handle.Bytes()), binary.LittleEndian, &fh)
```

Comment:
Added comment. Doesn't appear there is a `binary.NativeEndian`, see: https://groups.google.com/forum/#!topic/golang-nuts/3GEzwKfRRQw

@fspmarshall Since all the platforms we support are little endian I propose we leave this as-is. What do you think?

Reply:
@russjones oh wow, that thread is depressing.  Anyhow, I'm fine with it. Can we add a note to the effect that this code assumes platform endianness?  Feels like the kind of thing that shouldn't go undocumented, regardless of our currently supported platforms.

Reply:
Done.